packages/comms/utils/index.ts

```diff
@@ -0,0 +1,253 @@
+                const primitiveType = operation[prop].replace(/xsd:/gi, "");
+                if (prop.indexOf("[]") > 0) {
+                    typeDefn[prop.slice(0, -2)] = primitiveType + "[]";
+                } else if (operation[prop].match(/.*\|.*\|.*/)) {
+                    const enumTypeName = operation[prop].split("|")[0]
+                    const { type, enumType, values } = parseEnum(operation[prop], types[enumTypeName]);
+                    parsedEnums[type] = values;
+                    typeDefn[prop] = type;
+                } else {
+                    typeDefn[prop] = primitiveType;
+                }
+                if (Object.keys(primitiveMap).indexOf(primitiveType) > -1 && knownPrimitives.indexOf(primitiveType) < 0) {
+                    knownPrimitives.push(primitiveType);
+                }
+            }
+        }
+    }
+
+    if (knownTypes.indexOf(opName) < 0) {
+        knownTypes.push(opName);
+        parsedTypes[opName] = typeDefn;
+    }
+    return typeDefn;
+}
+
+wsdlToTs(args.url)
+    .then(clientObjs => {
+        const [wsdl, descr] = clientObjs;
+        const bindings = wsdl.definitions.bindings;
+        const wsdlNS = wsdl.definitions.$targetNamespace;
+        let namespace = "";
+        for (const ns in descr) {
+            namespace = changeCase(ns, Case.PascalCase);
+            const service = descr[ns];
+            printDbg("namespace: ", namespace, "\n");
+            for (const op in service) {
+                printDbg("binding: ", changeCase(op, Case.PascalCase), "\n");
+                const binding = service[op];
+                for (const svc in binding) {
+                    const operation = binding[svc];
+                    const types = wsdl.definitions.schemas[wsdlNS].types;
+                    const request = operation["input"];
+                    const reqName = bindings[op].methods[svc].input.$name;
+                    const response = operation["output"];
+                    const respName = bindings[op].methods[svc].output.$name;
+
+                    parseTypeDefinition(request, reqName, types);
+                    parseTypeDefinition(response, respName, types);
+                }
+            }
+        }
+
+        const relativePath = path.relative(path.join(cwd, outDir), path.join(cwd, "./src")).replace(/\\/g, "/");
+        lines.push(`import { IConnection, IOptions } from "${relativePath}/connection";`);
+        lines.push(`import { Service } from "${relativePath}/espConnection";`);
+        lines.push("\n\n");
+
+        knownPrimitives.forEach(primitive => {
+            lines.push(`type ${primitive} = ${primitiveMap[primitive]};`);
+        });
+        lines.push("\n\n");
+
+        for (const name in parsedEnums) {
+            lines.push(`export enum ${name} {\n\n`);
+            lines.push(parsedEnums[name].join(",\n"));
+            lines.push("\n\n}");
+            lines.push("\n\n");
+        }
```

Comment:
Minor, but the following might be easier to read (and the same for your other "templates"?:
```
lines.push(`\
export enum ${name} {
  ${parsedEnums[name].join(",\n  ")}
}

`);
```